LCORE-38: document BYOK (#92380)
openshift/openshift-docs: modules/ols-byo-knowledge-overview.adoc

// Module included in the following assemblies:
// * lightspeed-docs-main/configure/ols-configuring-openshift-lightspeed.adoc

:_mod-docs-content-type: CONCEPT
[id="byo-knowledge-overview_{context}"]

> = BYO Knowledge overview

Reviewer: Consider
```suggestion
= About the BYO Knowledge tool
```

> The {ols-long} service uses a large language model (LLM) to generate responses to questions. The BYO Knowledge tool can be used to create a retrieval-augmented generation (RAG) database to enhance the knowledge that is available to the LLM. When you create a RAG database you customize the {ols-long} service for your environment. For example, a network administrator can develop a standard operating procedure (SOP) that is used to provision an {ocp-product-title} cluster. Then, the network administrator can use the BYO Knowledge tool to enhance the knowledge available to the LLM by including information from the SOP.

Reviewer: Is the tool title always abbreviated, or is it more officially "the Bring Your Own (BYO) Knowledge tool" or somesuch? IOW, does it need to be expanded on first appearance?

```suggestion
The {ols-long} service uses a large language model (LLM) to generate responses to questions. You can enhance the knowledge that is available to the LLM by using the BYO Knowledge tool to create a retrieval-augmented generation (RAG) database. 

When you create a RAG database, you customize the {ols-long} service for your environment. For example, a network administrator can develop a standard operating procedure (SOP) that is used to provision an {ocp-product-title} cluster. Then, the network administrator can use the BYO Knowledge tool to enhance the knowledge available to the LLM by including information from the SOP.
```

Author: done

> To bring your own knowledge to an LLM:

Reviewer: Feel free to rephrase the specifics but all list lead ins must be complete sentences (it's a localization issue)
```suggestion
To bring your own knowledge to an LLM, you complete the following steps:
```

Author: done

* Create the custom content in Markdown format

> * Using the BYO Knowledge tool, package the content as a container image

Reviewer: ```suggestion
* Use the BYO Knowledge tool to package the content as a container image
```

Author: done

> * Push the newly created container image to an image registry, such as `quay.io`

Reviewer: ```suggestion
* Push the container image to an image registry, such as `quay.io`
```

Author: done

> * Update the `OLSConfig` custom resource file to list the image that was pushed to the image registry

Reviewer: ```suggestion
* Update the `OLSConfig` custom resource file to list the image that you pushed to the image registry
```

Author: done

> * Access the {ols-long} virtual assistant and submit a question associated with the custom knowledge that was added to the LLM.

Reviewer: ```suggestion
* Access the {ols-long} virtual assistant and submit a question that is associated with the custom knowledge that you added to the LLM.
```
Also, this one ends in a `.`, the rest should too

Author: done
